internal/checker/terraform/rules/azure/verify_azure_storage_minimum_tls_version.rego

```diff
@@ -0,0 +1,53 @@
+# METADATA
+# title: "Verify Storage Account is using the latest version of TLS encryption"
+# description: "This policy is designed to ensure the security and data protection of Azure Storage Accounts by verifying that the latest version of Transport Layer Security (TLS) encryption is in use."
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account
+# custom:
+#   id: CB_TFAZR_012
+#   severity: MEDIUM
+package lib.terraform.CB_TFAZR_012
+
+isvalid(block) {
+	block.Type == "resource"
+	block.Labels[_] == "azurerm_storage_account"
+}
+
+resource[resource] {
+	block := pass[_]
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	block := fail[_]
+	resource := concat(".", block.Labels)
+}
+
+pass[block] {
+	block := input[_]
+	isvalid(block)
```

Comment:
add supported TLS like this
validValue := ["TLS1_2", "TLS1_3"]
lock.Attributes.min_tls_version == validValue[_]